contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -526,15 +528,24 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         return receiptQueues[originBlockchainID].getReceiptAtIndex(index);
     }
 
+    /**
+     * @dev Calculates the message ID for the given source blockchain ID and message nonce.
+     */
+    function calculateMessageID(
+        bytes32 sourceBlockchainID,
+        bytes32 destinationBlockchainID,
```

Comment:
I wonder if it would be worth trying to distinguish these two using different type aliases.

Reply:
That's an interesting idea. I think it requires it's own investigation outside the scope of this issue though, so I'm going to leave it as is for now.